### ISSUE TITLE
SNOW-1250044: Available anaconda packages aren't always detected

### ISSUE THREAD

**Issue by sfc-gh-hachouraria** [snowflakedb/snowflake-cli#914]
### SnowCLI version

2.1.0

### Python version

3.11

### Platform

macOS 14.4 (Intel)

### What happened

Ran `snow snowpark package create fuzzymatcher -y --verbose` to be able to use a `fuzzymatcher` zip through a stage package.

The package `fuzzymatcher` depends, amongst others, on `rapidfuzz` which is on Snowflake's Anaconda channel as seen at: https://repo.anaconda.com/pkgs/snowflake/

However, the command did not detect that `rapidfuzz` must be consumed through Anaconda, and instead built a zip with `rapidfuzz` and its inner dependencies that include shared libraries.

### Console output

```Shell
~> snow snowpark package create fuzzymatcher -y --verbose
2024-03-19 08:03:56 INFO 'fuzzymatcher' not found in Snowflake Anaconda channel...
2024-03-19 08:04:12 INFO Checking for dependencies available in Anaconda...
2024-03-19 08:04:12 INFO Opening file .packages/tzdata-2024.1.dist-info/METADATA in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/tzdata-2024.1.dist-info/METADATA
2024-03-19 08:04:12 INFO Opening file .packages/tzdata-2024.1.dist-info/RECORD in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/tzdata-2024.1.dist-info/RECORD
2024-03-19 08:04:12 INFO Opening file .packages/six-1.16.0.dist-info/METADATA in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/six-1.16.0.dist-info/METADATA
2024-03-19 08:04:12 INFO Opening file .packages/six-1.16.0.dist-info/RECORD in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/six-1.16.0.dist-info/RECORD
2024-03-19 08:04:12 INFO Opening file .packages/pytz-2024.1.dist-info/METADATA in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/pytz-2024.1.dist-info/METADATA
2024-03-19 08:04:12 INFO Opening file .packages/pytz-2024.1.dist-info/RECORD in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/pytz-2024.1.dist-info/RECORD
2024-03-19 08:04:12 INFO Opening file .packages/Levenshtein-0.25.0.dist-info/METADATA in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/Levenshtein-0.25.0.dist-info/METADATA
2024-03-19 08:04:12 INFO Opening file .packages/Levenshtein-0.25.0.dist-info/RECORD in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/Levenshtein-0.25.0.dist-info/RECORD
2024-03-19 08:04:12 INFO Opening file .packages/Metaphone-0.6.dist-info/METADATA in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/Metaphone-0.6.dist-info/METADATA
2024-03-19 08:04:12 INFO Opening file .packages/Metaphone-0.6.dist-info/RECORD in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/Metaphone-0.6.dist-info/RECORD
2024-03-19 08:04:12 INFO Opening file .packages/python_dateutil-2.9.0.post0.dist-info/METADATA in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/python_dateutil-2.9.0.post0.dist-info/METADATA
2024-03-19 08:04:12 INFO Opening file .packages/python_dateutil-2.9.0.post0.dist-info/RECORD in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/python_dateutil-2.9.0.post0.dist-info/RECORD
2024-03-19 08:04:12 INFO Opening file .packages/numpy-1.24.4.dist-info/METADATA in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/numpy-1.24.4.dist-info/METADATA
2024-03-19 08:04:12 INFO Opening file .packages/numpy-1.24.4.dist-info/RECORD in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/numpy-1.24.4.dist-info/RECORD
2024-03-19 08:04:12 INFO Opening file .packages/rapidfuzz-3.6.2.dist-info/METADATA in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/rapidfuzz-3.6.2.dist-info/METADATA
2024-03-19 08:04:12 INFO Opening file .packages/rapidfuzz-3.6.2.dist-info/RECORD in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/rapidfuzz-3.6.2.dist-info/RECORD
2024-03-19 08:04:12 INFO Opening file .packages/fuzzymatcher-0.0.6.dist-info/METADATA in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/fuzzymatcher-0.0.6.dist-info/METADATA
2024-03-19 08:04:12 INFO Opening file .packages/fuzzymatcher-0.0.6.dist-info/RECORD in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/fuzzymatcher-0.0.6.dist-info/RECORD
2024-03-19 08:04:12 INFO Opening file .packages/pandas-2.0.3.dist-info/METADATA in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/pandas-2.0.3.dist-info/METADATA
2024-03-19 08:04:12 INFO Opening file .packages/pandas-2.0.3.dist-info/RECORD in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/pandas-2.0.3.dist-info/RECORD
2024-03-19 08:04:12 INFO Opening file .packages/python_Levenshtein-0.25.0.dist-info/METADATA in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/python_Levenshtein-0.25.0.dist-info/METADATA
2024-03-19 08:04:12 INFO Opening file .packages/python_Levenshtein-0.25.0.dist-info/RECORD in mode 'r'
2024-03-19 08:04:12 INFO Closing file .packages/python_Levenshtein-0.25.0.dist-info/RECORD
2024-03-19 08:04:12 INFO Downloaded packages: dict_keys(['tzdata', 'six', 'pytz', 'Levenshtein', 'Metaphone', 'python-dateutil', 'numpy', 'rapidfuzz', 'fuzzymatcher', 'pandas', 'python-Levenshtein'])
2024-03-19 08:04:19 INFO 'tzdata' not found in Snowflake Anaconda channel...
2024-03-19 08:04:19 INFO 'pytz' not found in Snowflake Anaconda channel...
2024-03-19 08:04:19 INFO 'Levenshtein' not found in Snowflake Anaconda channel...
2024-03-19 08:04:19 INFO 'Metaphone' not found in Snowflake Anaconda channel...
2024-03-19 08:04:19 INFO 'python-dateutil' not found in Snowflake Anaconda channel...
2024-03-19 08:04:19 INFO 'rapidfuzz' not found in Snowflake Anaconda channel...
2024-03-19 08:04:19 INFO 'fuzzymatcher' not found in Snowflake Anaconda channel...
2024-03-19 08:04:19 INFO 'python-Levenshtein' not found in Snowflake Anaconda channel...
2024-03-19 08:04:19 INFO Good news! The following package dependencies can be
                imported directly from Anaconda, and will be excluded from
                the zip: SplitRequirements(snowflake=[<Requirement: "six==1.16.0">, <Requirement: "numpy==1.24.4">, <Requirement: "pandas==2.0.3">], other=[<Requirement: "tzdata==2024.1">, <Requirement: "pytz==2024.1">, <Requirement: "Levenshtein==0.25.0">, <Requirement: "Metaphone==0.6">, <Requirement: "python-dateutil==2.9.0.post0">, <Requirement: "rapidfuzz==3.6.2">, <Requirement: "fuzzymatcher==0.0.6">, <Requirement: "python-Levenshtein==0.25.0">])
2024-03-19 08:04:19 INFO Package six: deleting 3 files
2024-03-19 08:04:19 INFO Removing directory .packages/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/six-1.16.0.dist-info
2024-03-19 08:04:19 INFO Removing file .packages/six.py
2024-03-19 08:04:19 INFO Package numpy: deleting 127 files
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/ma/tests/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/f2py/tests/src/quoted_character
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/typing/tests/data/fail
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/lib/tests/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/random/tests
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/f2py/tests/src/array_from_pyobj
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/array_api/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/f2py/tests/src/parameter
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/f2py/tests/src/abstract_interface
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/random
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/distutils/checks
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/f2py
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/fft/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/_pyinstaller
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/core/tests/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/typing
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/polynomial
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/core/tests/examples/cython
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/core/tests/data
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/core/tests
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/core/include/numpy
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/core/lib/npy-pkg-config
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/ma/tests
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/distutils/command/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/distutils/fcompiler
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/testing/tests
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/distutils/tests/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/compat/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/tests/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/array_api/tests/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy-1.24.4.dist-info
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/compat/tests/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/ma
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/distutils
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/testing
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/_typing
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/array_api/tests
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/core/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/fft/tests/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/linalg/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/core
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/doc/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/fft
2024-03-19 08:04:19 INFO Removing directory .packages/numpy/matrixlib/tests/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/numpy
2024-03-19 08:04:19 INFO Package pandas: deleting 304 files
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/config/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/indexes/period
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/indexing
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/io/xml/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/io/formats/style
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/scalar
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/core/dtypes/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/arrays/period/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/base/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/tslibs/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/reshape/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/groupby/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/core/interchange
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/arrays/timedeltas
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/indexes/numeric
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/libs/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/tests/plotting/frame/__pycache__
2024-03-19 08:04:19 INFO Removing directory .packages/pandas/core/sparse
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/extension
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/io/formats/templates
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/frame
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/window/moments/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/core/indexes
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/_testing
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/core/_numba/kernels
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/indexes/interval
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/tslibs
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/io
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/_libs/tslibs/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/core/indexers/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/indexes/multi/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/core/window/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/core/window
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/arrays/sparse/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/computation/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/io/xml
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/core/arrays/sparse
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/io/pytables
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/arrays/numpy_/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/util
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/internals/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/arrays/sparse
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/io/data/xml
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/strings
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/indexes/datetimes/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/io/data
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/arrays/interval
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/groupby
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/arrays/string_
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/core/_numba
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/interchange/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/io/parser/usecols
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/libs
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/window/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/tools/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/core/methods
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/arrays/masked/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/computation
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/copy_view/index/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/core/ops
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/internals
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/arrays
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/tseries/holiday/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/tseries/frequencies/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/io/parser
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/core/computation
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/reshape/concat
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/indexes
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/api/interchange
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/series/accessors
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/resample/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/util
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/reshape
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/errors
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/compat/numpy
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tseries/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/_libs/window
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/copy_view/index
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/series/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/arithmetic/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/arrays/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests/io/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/_libs/__pycache__
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/_libs
2024-03-19 08:04:20 INFO Removing directory .packages/pandas/tests
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/__pycache__
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/indexers
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/internals/__pycache__
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/array_algos
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/_config/__pycache__
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/tools/__pycache__
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/api/indexers/__pycache__
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/tools
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/strings
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/compat/__pycache__
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/api/types
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/arrays/arrow/__pycache__
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/groupby
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/plotting/_matplotlib
2024-03-19 08:04:21 INFO Removing directory .packages/pandas-2.0.3.dist-info
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/internals
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/dtypes
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/util/__pycache__
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/api
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/arrays
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/plotting
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/arrays/arrow
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/util
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core/reshape
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/core
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/tseries
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/compat
2024-03-19 08:04:21 INFO Removing directory .packages/pandas/_config
2024-03-19 08:04:21 INFO Removing directory .packages/pandas
2024-03-19 08:04:21 INFO Checking to see if packages have native libraries...
2024-03-19 08:04:21 INFO Potential native library: .packages/Levenshtein/levenshtein_cpp.cpython-38-darwin.so
2024-03-19 08:04:21 INFO Potential native library: .packages/rapidfuzz/process_cpp_impl.cpython-38-darwin.so
2024-03-19 08:04:21 INFO Potential native library: .packages/rapidfuzz/_feature_detector_cpp.cpython-38-darwin.so
2024-03-19 08:04:21 INFO Potential native library: .packages/rapidfuzz/fuzz_cpp_avx2.cpython-38-darwin.so
2024-03-19 08:04:21 INFO Potential native library: .packages/rapidfuzz/fuzz_cpp.cpython-38-darwin.so
2024-03-19 08:04:21 INFO Potential native library: .packages/rapidfuzz/utils_cpp.cpython-38-darwin.so


WARNING! Some packages appear to have native libraries!
Continue with package installation? [y/N]: y
2024-03-19 08:06:10 INFO Removing directory .packages
Package fuzzymatcher.zip created. You can now upload it to a stage (`snow snowpark package upload -f fuzzymatcher.zip -s <stage-name>`) and reference it in your procedure or function.
The package fuzzymatcher is supported, but does depend on the
                following Snowflake supported libraries. You should
                include the following in your packages:
                six==1.16.0
numpy==1.24.4
pandas==2.0.3
```

### How to reproduce

Run:

```
snow snowpark package create fuzzymatcher -y --verbose
```

Expected: 

Hopefully a pure Python zip with only entries corresponding to packages missing from the Anaconda channel.

Observed: 

A mix of native code shared objects and pure Python entries including several packages that are already in Anaconda channel.

Attached:

The unusable zip as created by Snowflake CLI 2.1.0 and command above: 
[fuzzymatcher-do-not-use.zip](https://github.com/snowflakedb/snowflake-cli/files/14644641/fuzzymatcher.zip)

**Comment by sfc-gh-pjob** [comment]
It looks like https://github.com/snowflakedb/snowflake-cli/issues/763 and it should be already fixed on the main branch and should be released in Snowflake CLI 2.2.0.

Size of ZIP file created by the code on the main branch:
```
15M Mar 20 17:45 fuzzymatcher.zip
```

And size of ZIP file created by v2.1.0:
```
26M Mar 20 16:45 fuzzymatcher.zip
```

However this particular case still won't work as expected because of a bug in dependency version comparison logic. We don't handle correctly dependencies which specify both min and max version (we assume that there is only min version). And as the result, even on the main branch, numpy and rapidfuzz land in the ZIP.

https://github.com/snowflakedb/snowflake-cli/blob/381ff2aa1a9b52cb950ccff6e75e2d2f80f70824/src/snowflake/cli/plugins/snowpark/package/anaconda.py#L27
<img width="478" alt="Screenshot 2024-03-20 at 17 55 02" src="https://github.com/snowflakedb/snowflake-cli/assets/132934591/21c71493-a024-4124-8976-7b651b7b720e">

-- 

We'll work on the fix and keep you updated but it will be probably in 2.2.0 release.

**Comment by sfc-gh-pczajka** [comment]
This have been fixed and will be released in 2.2.0

**Comment by sfc-gh-pjob** [comment]
The fix can be tested using the release candidate (pre-release) for 2.2.0 which can be installed the following way:

```
pip install snowflake-cli-labs==2.2.0rc0
```

However, as it is only the release candidate, it should be used only for tests, not in the production.

**Comment by sfc-gh-pjob** [comment]
We released CLI v2.2.0. The fix is there.
